source/utilities.cc

```diff
@@ -745,3 +745,65 @@ namespace aspect
     template std_cxx1x::array<double,3> spherical_coordinates<3>(const Point<3> &position);
   }
 }
+
+namespace dealii
+{
+  namespace Utilities
+  {
+    namespace MPI
+    {
+      template<int dim>
+      Tensor<1,dim> 
+      reduce_tensor(const Tensor<1,dim> &local, const MPI_Comm &mpi_communicator)
+      {
+        std::vector<double> entries(dim);
+        std::vector<double> global_entries(dim);
+        Tensor<1,dim> global;
```

Comment:
Use fixed-sized arrays `double entries[dim]` here since you know the size and don't need dynamic memory allocation. Move the declaration of `global` down to right above the first use. Restricting the live range of variables makes it so much easier to optimize for compilers.